From f6d2af38f52af14f4baec39ee3695041b4692eb0 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 20 Jun 2024 13:39:55 +0300
Subject: [PATCH] [FSW-2704] fs1: add manifest spi dma constants

SPI DMA of manifests is from fixed offset in SPI
to fixed offsets in ICCM. This patch defines all
constant values needed for copying both manifest
0 and 1.
Manifest0 : SRC SPI offset: 0x0 Dest ICCM Offset: 0x0
Manifest1 : SRC SPI offset: 0x400000 Dest ICCM Offset: 0x10000

Change-Id: I7a229c1d4beff63fedcb4fb0f69c189203e85f99
---

diff --git a/arch/fs1/arc/soc.h b/arch/fs1/arc/soc.h
index a60d690..1037c0c 100644
--- a/arch/fs1/arc/soc.h
+++ b/arch/fs1/arc/soc.h
@@ -96,4 +96,20 @@
 #define HL_FW_SRAM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_FWSRAM_OFFSET)
 #define HL_FW_SRAM_SIZE			SZ_128K
 
+#define SPI_MANIFEST_IMG_0_FLASH_OFFSET		(0x0)
+/* manifest 0 boot pointers are DMA trans to ARC1 ICCM at offset 0 */
+#define SPI_MANIFEST_IMG_0_ICCM_OFFSET		(0)
+#define SPI_MANIFEST_IMG_0_OFFSET		(HL_ICCM_OFFSET + SPI_MANIFEST_IMG_0_ICCM_OFFSET)
+#define SPI_MANIFEST_IMG_0_DMA_DEST		(HL_FW_MEM_BASE_FULL + SPI_MANIFEST_IMG_0_OFFSET)
+/* 32Bit arc viewpoint address of manifest 0 stored in ICCM */
+#define SPI_MANIFEST_IMG_0_DMA_DEST_ARC		(HL_FW_MEM_BASE + SPI_MANIFEST_IMG_0_OFFSET)
+
+#define SPI_MANIFEST_IMG_1_FLASH_OFFSET		(0x200000)
+/* manifest 1 boot pointers are DMA trans to ARC1 ICCM at offset 64K */
+#define SPI_MANIFEST_IMG_1_ICCM_OFFSET		(64 * 1024)
+#define SPI_MANIFEST_IMG_1_OFFSET		(HL_ICCM_OFFSET + SPI_MANIFEST_IMG_1_ICCM_OFFSET)
+#define SPI_MANIFEST_IMG_1_DMA_DEST		(HL_FW_MEM_BASE_FULL + SPI_MANIFEST_IMG_1_OFFSET)
+/* 32Bit arc viewpoint address of manifest 1 stored in ICCM */
+#define SPI_MANIFEST_IMG_1_DMA_DEST_ARC		(HL_FW_MEM_BASE + SPI_MANIFEST_IMG_1_OFFSET)
+
 #endif /* _SOC_H_ */
